iothub/service/src/Exceptions/IotHubServiceException.cs

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Net;
+using System.Runtime.Serialization;
+
+namespace Microsoft.Azure.Devices.Common.Exceptions
+{
+    /// <summary>
+    /// The exception thrown when the client receives an error while communicating with IoT hub service.
+    /// </summary>
+    [Serializable]
+    public class IotHubServiceException : Exception
+    {
+        private const string IsTransientValueSerializationStoreName = "IotHubServiceException-IsTransient";
+        private const string TrackingIdSerializationStoreName = "IoTHubException-TrackingId";
+
+        /// <summary>
+        /// Creates an instance of <see cref="IotHubServiceException"/> with the supplied error message and marks it as non-transient.
```

Comment:
No need to reference this type in doc comments.

```suggestion
        /// Creates an instance of this class with the specified error message and optional inner exception.
```